### PR TITLE
reduce rbac rules

### PR DESCRIPTION
**Pull request by 27149chen**

### CODE REVIEW

File: deploy/role.yaml [hunk 1]
```diff
@@ -11,14 +11,47 @@ rules:
...
+  resources:
+  - configmaps
+  verbs:
+  - get
+  - create
+  - delete
+- apiGroups:
+  - ""
+  resources:
+  - secrets
```
Comment (shay-berman) [inline deploy/role.yaml]: why operator need to get secrets?

File: deploy/role.yaml [hunk 2]
```diff
@@ -11,14 +11,47 @@ rules:
...
   - pods
-  - services
-  - services/finalizers
-  - endpoints
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
```
Comment (shay-berman) [inline deploy/role.yaml]: why need to create configmap?

File: deploy/role.yaml [hunk 3]
```diff
@@ -11,14 +11,47 @@ rules:
...
+  - ""
+  resources:
   - persistentvolumes
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+  - create
+  - delete
```
Comment (shay-berman) [inline deploy/role.yaml]: why operator need to delete\create PV? its the driver that need that.
what do i miss?

Comment (27149chen) [reply deploy/role.yaml]: Please see my comment under the ticket first, as I said:

>  in order to create all the clusterroles of the driver, the operator's clusterrole needs all the permissions inside these clusterroles, please make sure the rules in the clusterroles of the driver is correct

File: deploy/role.yaml (hunk 2 above)
Comment (27149chen) [reply deploy/role.yaml]: operator will use it to create a lock for leader selection.

File: deploy/role.yaml (hunk 1 above)
Comment (27149chen) [reply deploy/role.yaml]: because ibm-block-csi-external-provisioner-role need it

File: deploy/role.yaml (hunk 2 above)
Comment (shay-berman) [reply deploy/role.yaml]: sure
sorry

File: deploy/role.yaml (hunk 1 above)
Comment (shay-berman) [reply deploy/role.yaml]: so why it should be here?
this is the operator deployment and not the csi-controller.   do i miss something?

Comment (27149chen) [reply deploy/role.yaml]: as mentioned below, the operator role needs all the rules of the driver, including the csi-external-provisioner

Comment (27149chen) [reply deploy/role.yaml]: 1. Yes, I'm sure. It is easy to understand the reason: the role is to limit the permission of the operator, if I can create a new role in the operator and give more permissions, then there is a big security issue - in fact I can do anything!
4. Good question. I thought about it before. But the operator role has more permissions than the driver, if we use the operator role for the driver, the driver will have too big power.

Comment (shay-berman) [reply deploy/role.yaml]: thanks for the answers
its clear now
Base on your detail it LGTM

**Pull request merged**